ingest/src/main/java/com/microsoft/azure/kusto/ingest/IngestClientImpl.java

```diff
@@ -181,17 +181,22 @@ public IngestionResult ingestFromStream(StreamSourceInfo streamSourceInfo, Inges
                     ingestionProperties.getDatabaseName(),
                     ingestionProperties.getTableName(),
                     ingestionProperties.getDataFormat(),
-                    streamSourceInfo.getCompressionType());
+                    streamSourceInfo.getCompressionType() != null ? streamSourceInfo.getCompressionType() : CompressionType.gz);
```

Comment:
This condition should count shouldCompress as well (otherwise we will add redundant .gz extension when its parquet or orc streams).